internal/fsx/fsx_test.go

```diff
@@ -0,0 +1,74 @@
+package fsx_test
+
+import (
+	"bufio"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/ooni/probe-engine/internal/fsx"
+)
+
+var StateBaseDir = "../../testdata/fsx/"
+
+func TestFileInputInvalidFile(t *testing.T) {
+	dir, err := ioutil.TempDir(StateBaseDir, "test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(dir)
+
+	// test function panics with invalid file
+	defer func() { recover() }()
+
+	dir += "/test.txt"
+	_, err = fsx.Open(dir)
+	if err == nil {
+		t.Errorf("expected panic with invalid input file")
+	}
+
+}
+
+func TestFileInputDirectory(t *testing.T) {
+	dir, err := ioutil.TempDir(StateBaseDir, "test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(dir)
+
+	// test function panics with invalid directory
+	defer func() { recover() }()
+
+	_, err = fsx.Open(dir)
+	if err == nil {
+		t.Errorf("expected panic with input directory")
+	}
+}
+
+func TestInputFile(t *testing.T) {
+	// create test input file
+	input := "my input 1"
+	file_contents := []byte(input)
+	inputfile, err := ioutil.TempFile(StateBaseDir, "myinput")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(inputfile.Name())
+
+	if _, err := inputfile.Write(file_contents); err != nil {
+		t.Fatal(err)
+	}
+	if err := inputfile.Close(); err != nil {
+		t.Fatal(err)
+	}
+
```

Comment:
```suggestion
```

Please create a file in `./internal/fsx/testdata` and commit it. So we don't need to code above and we can just open such file in our test, resulting in a shorter and easier to understand test implementation.